File: Arrays101/Deleting Item From Array/removeduplicatesfromsortedarray.py

nums = [0,0,1,1,1,2,2,3,3,4]
def removeDuplicates(nums):
    i = 0
    j = 1
    
    while j< len(nums):
        if nums[i] != nums[j]:
            i += 1
            nums[i] = nums[j]
        j += 1 
    return i + 1       
removeDuplicates(nums)

File: Arrays101/RemoveDuplicatesfromSortedArray.java
package Arrays101;

public class RemoveDuplicatesfromSortedArray {
    public static void main(String[] args) {
        int[] temp = { 1, 1, 2,2,3,3,4,4,4 };
        removeDuplicates(temp);
    }

    public static int removeDuplicates(int[] nums) {

        int j = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            if(nums[i] != nums[i + 1]){
                nums[j++] = nums[i];
            }
        }
        nums[j++] = nums[nums.length -1];

        return j;
    }
}
